x-pack/plugins/security/server/session_management/session_index.ts

```diff
@@ -0,0 +1,397 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { ILegacyClusterClient, Logger } from '../../../../../src/core/server';
+import { AuthenticationProvider } from '../../common/types';
+import { ConfigType } from '../config';
+
+export interface SessionIndexOptions {
+  readonly clusterClient: ILegacyClusterClient;
+  readonly indexName: string;
+  readonly config: Pick<ConfigType, 'session'>;
+  readonly logger: Logger;
+}
+
+/**
+ * Alias of the Elasticsearch index that is used to store user session information.
+ */
+export const SESSION_INDEX_ALIAS = '.kibana_security_session';
+
+/**
+ * Version of the current session index template.
+ */
+export const SESSION_INDEX_TEMPLATE_VERSION = 1;
+
+/**
+ * Name of the current session index template.
+ */
+export const SESSION_INDEX_TEMPLATE_NAME = `${SESSION_INDEX_ALIAS}_index_template_${SESSION_INDEX_TEMPLATE_VERSION}`;
+
+/**
+ * Returns index template that is used for the current version of the session index.
+ */
+export function getSessionIndexTemplate(indexName: string) {
+  return Object.freeze({
+    index_patterns: indexName,
+    order: 1000,
+    settings: {
+      index: {
+        number_of_shards: 1,
+        number_of_replicas: 0,
+        auto_expand_replicas: '0-1',
+        priority: 1000,
+        refresh_interval: '1s',
+        hidden: true,
+      },
+    },
+    mappings: {
+      dynamic: 'strict',
+      properties: {
+        usernameHash: { type: 'keyword' },
+        provider: { properties: { name: { type: 'keyword' }, type: { type: 'keyword' } } },
+        idleTimeoutExpiration: { type: 'date' },
+        lifespanExpiration: { type: 'date' },
+        accessAgreementAcknowledged: { type: 'boolean' },
+        content: { type: 'binary' },
+      },
+    },
+    aliases: { [SESSION_INDEX_ALIAS]: {} },
```

Comment:
I think we need a distinct alias per tenant. Otherwise, the alias will end up resolving to all session indices, including those that don't belong to the current tenant

Reply:
__note:__ we discussed this offline and decided to get rid of session index alias completely as there is no need in it currently (handled in https://github.com/elastic/kibana/pull/68117/commits/a8d46f171c0ecf328600ce94842fb7a37d8fdb3f).